script/check-patch-diff.js

```diff
@@ -0,0 +1,40 @@
+const { execSync, spawnSync } = require('child_process');
+const path = require('path');
+const { inspect } = require('util');
+
+const srcPath = path.resolve(__dirname, '..', '..', '..');
+const patchExportPath = path.resolve(__dirname, 'export_all_patches.py');
+const configPath = path.resolve(__dirname, '..', 'patches', 'config.json');
+
+// Re-export all the patches to check if there were changes.
+execSync(`python ${patchExportPath} ${configPath}`, { cwd: srcPath });
+
+// Update indices.
+execSync('git update-index --refresh || true');
+
+// If the exit code is 0, then there were no differences.
+const { status } = spawnSync('git', ['diff-index', '--quiet', 'HEAD']);
+
+if (status !== 0) {
+  const committerName = execSync('git config user.name');
+  const committerEmail = execSync('git config user.email');
+
+  const env = Object.assign({}, process.env, {
+    GIT_COMMITTER_NAME: committerName || 'Electron Bot',
+    GIT_COMMITTER_EMAIL: committerEmail || 'anonymous@electronjs.org'
+  });
```

Comment:
I don't think there's any need for this bit here as it's running on a user's local machine, prior to commit. So we don't need to make a new commit inside of the lint step.